app/models/fleet_truck.rb

```diff
@@ -0,0 +1,11 @@
+class FleetTruck < ActiveRecord::Base
+
+  belongs_to :truck
+  belongs_to :fleet
+
+  # validates :truck, uniqueness: true
+  # validates_uniqueness_of :truck, :scope => [:registration]
+  validates :truck_id, uniqueness: {scope: :fleet_id} 
+  validates :status, presence: true
```

Comment:
You also need to validate that the status is one of the expected value (EG: "standard" or "auxiliary").